src/golang/cmd/server/handler/preview.go

```diff
@@ -75,6 +81,59 @@ func (*PreviewHandler) Name() string {
 	return "Preview"
 }
 
+func checkError(w http.ResponseWriter, err error) {
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+}
+
+func (*PreviewHandler) SendResponse(w http.ResponseWriter, response interface{}) {
+
+	structResponse := response.(previewResponse)
+
+	responseNoArtifacts := previewResponseNoArtifacts{
+		Status:          structResponse.Status,
+		OperatorResults: structResponse.OperatorResults,
+	}
+
+	jsonBlob, err := json.Marshal(responseNoArtifacts)
+	checkError(w, err)
+
+	mw := multipart.NewWriter(w)
+	w.Header().Set("Content-Type", mw.FormDataContentType())
+
+	fw, errA := mw.CreateFormField("metadata")
+	checkError(w, errA)
+
+	_, errB := fw.Write(jsonBlob)
+	checkError(w, errB)
+
+	for id, art := range structResponse.ArtifactResults {
+
+		art_content := art.Content
+		art_serialization := art.SerializationType
+		serialization_type := "serialization_type" + id.String()
+
+		fw, errC := mw.CreateFormField(id.String())
+		checkError(w, errC)
+
+		_, errD := fw.Write(art_content)
+		checkError(w, errD)
+
+		fw, errE := mw.CreateFormField(serialization_type)
+		checkError(w, errE)
+
+		_, errF := fw.Write([]byte(art_serialization))
+		checkError(w, errF)
+	}
```

Comment:
We shouldn't define variables with names like `errA`, ..., `errF`. Generally not good in terms of code cleanness and readability, and since each error won't be needed after the `checkError`, we can use the same `err` variable name for subsequent api calls.